### DISCUSSION TITLE
GitHub markdown for tables, first hline after second row

### DISCUSSION THREAD

**Discussion by nbehrnd** [community/community#53017] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

This question is still open after consultation of the [guidelines](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github#enabling-fixed-width-fonts-in-the-editor). I would like to write a table where the (first / only) horizontal line follows the second row.

The output on GitHub's display for

```
| command  | description  |
|--------  |------------  |
| alpha    | so and so    |
| beta     | so and so    | 
```

works well.  There however are instances, where content of the second row is more related to first column (e.g. by provision of a unit).  Hence, instead of an input of 

```
| item   | length | colour |
|------  |--------|----    |
|        |  in cm |        |
| pencil | 12     | green  |
```

leading to 

![snippet01](https://user-images.githubusercontent.com/19709084/232434241-42b60366-ec2a-4db1-951e-930506988875.png)

I would like to define a table like

```
| item    | length | colour |
|         |  in cm |        |
|------   |--------|----    |
| pencil  | 12     | green  |
```

which however yields

![snippet02](https://user-images.githubusercontent.com/19709084/232434769-062eae41-1344-448a-ade5-c3ecf58f0a38.png)

As tested, the simple omission of the hline in the markdown does not do good, an input of 

```
| item    | length | colour |
|         |  in cm |        |
| pencil  | 12     | green  |
```

only provides

![snippet03](https://user-images.githubusercontent.com/19709084/232435234-d81dea67-87f1-4cde-80cc-aa1b7dab7381.png)

It is a bit surprising for me, because an orgmode table which would be just fine with 

``` org
| item    | length | colour |
|         |  in cm |        |
|---------+--------+--------|
| pencil  | 12     | green  |
```

is converted by [pandoc](https://pandoc.org/try/) (use from org, to markdown_github) to

```
|        |        |        |
|--------|--------|--------|
| item   | length | colour |
|        | in cm  |        |
| pencil | 12     | green  |
```

- note the lifted hline, and the additional empty row on top - to yield

![snippet04](https://user-images.githubusercontent.com/19709084/232436984-75c22bca-96ab-4bb4-bf33-f6e4a7d1e1cd.png)

including the empty line (blue mark).  Indeed, if this extra line is removed and I provide only

```
|--------|--------|--------|
| item   | length | colour |
|        | in cm  |        |
| pencil | 12     | green  |
```

the output is 

![snippet05](https://user-images.githubusercontent.com/19709084/232437741-298c34a8-b4be-4d55-9af6-8d24e5de7bde.png)

It is possible to request pandoc to provide an output as `markdown_strict`, which is

```
<table>
<tbody>
<tr class="odd">
<td>item</td>
<td>length</td>
<td>colour</td>
</tr>
<tr class="even">
<td></td>
<td>in cm</td>
<td></td>
</tr>
<tr class="odd">
<td>pencil</td>
<td>12</td>
<td>green</td>
</tr>
</tbody>
</table>
```

however at the expense of easy reading the source code to yield

![markdown_strict](https://user-images.githubusercontent.com/19709084/232438740-25366dfd-b072-4640-b48f-6f9404efcb32.png)

Regarding the encoding of the table, the outputs passing from orgmode to either markdown_github, or markdown_mmd, or markdown_phpextra are equal.

**Question:** What is the recommended approach to remove said not-so-nice extra empty row?  Is it perhaps something in the underlying  syntax in GitHub's definition of markdown which would require an adjustment?  Because if I export said orgmode table

``` orgmode

| item    | length | colour |
|         |  in cm |        |
|---------+--------+--------|
| pencil  | 12     | green  |
```

by Emacs (GNU Emacs 28.1) and command `C-c C-e h o` to pass it over to html, the pattern simply matches the anticipation:

![by_Emacs_orgmode](https://user-images.githubusercontent.com/19709084/232441065-34bc95bd-afa5-48c1-a600-7ed2bc0e6619.png)

**Comment by neutrinomuon** [comment, marked as answer]
Hi @nbehrnd, Have you tried just simply using the following format below? It looks like it works well:

    | item   | length <br>in cm | colour |
    |--------|:----------------:|--------|
    | pencil | 12.0             | green  |
    | pen    | 15.5             | blue   |

| item   | length <br>in cm | colour |
|--------|:----------------:|--------|
| pencil | 12.0             | green  |
| pen    | 15.5             | blue   |

I hope this helps, please let me know

**Comment by nbehrnd** [comment]
The suggestion works well for "the data already are in markdown".  I still have to identify an adjusted workflow if the source of the document is written in orgmode in first place; where - for the conversion into *github* markdown, pandoc is my goto.

Contrasting to initial speculation, a simple insertion of `<br>` into a cell (because orgmode's line feed `\\` affects running text only) will not work, apparently neither to Emacs orgmode itself, nor to pandoc.  A subsequent lesson was that beside "standard tables" there are tables by Emacs' package `table.el`, which actually may extend over multiple lines (a brief [how to](https://emacs.stackexchange.com/questions/38135/wrap-cell-content-in-an-org-mode-table)).  There is a subtle difference in their syntax (`+` at the cells vertices not only for cells within a table, but equally on the envelope) compared to the standard table, something which pandoc doesn't trade well (for now).  I think questions about and around *these points* are addressed better to the mailing lists of pandoc and/or orgmode.

![org2md_by_pandoc](https://user-images.githubusercontent.com/19709084/232525467-f34b1960-52e5-4877-a6be-26295ae4c358.png)

![org2html_in_Emacs_orgmode](https://user-images.githubusercontent.com/19709084/232525504-84b8bc17-8bbe-4aad-8998-9da4d53ddfd7.png)
